Author: Owen Kirby <okirby@mozilla.com>
Description: Fix QRegularExpression compilation on GCC 7.5
Forwarded: no
Last-Update: 2022-02-10

--- a/src/corelib/text/qregularexpression.cpp
+++ b/src/corelib/text/qregularexpression.cpp
@@ -1185,7 +1185,7 @@
     // subject may have data == nullptr. In this case, to keep PCRE
     // happy, pass a pointer to a dummy character.
     const char16_t dummySubject = 0;
-    const char16_t * const subjectUtf16 = [&]()
+    const char16_t * const subjectUtf16 = [priv, subjectLength, &dummySubject]()
     {
         const auto subjectUtf16 = priv->subject.utf16();
         if (subjectUtf16)
